pkg/services/extsvcauth/oauthserver/oasimpl/service.go

```diff
@@ -134,44 +137,51 @@ func (s *OAuth2ServiceImpl) GetExternalService(ctx context.Context, id string) (
 		return nil, err
 	}
 
-	// Handle the case where the external service has no service account
+	if err := s.setClientUser(ctx, client); err != nil {
+		return nil, err
+	}
+
+	s.cache.Set(id, *client, cacheExpirationTime)
+	return client, nil
+}
+
+// setClientUser sets the SignedInUser and SelfPermissions fields of the client
+func (s *OAuth2ServiceImpl) setClientUser(ctx context.Context, client *oauthserver.OAuthExternalService) error {
```

Comment:
I extracted that function out of `GetExternalService` so I could reuse it in `handlePluginStateChange` to fetch the `SelfPermissions`